org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/cvc_elt_1_aCodeAction.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.util.List;
+import java.util.Collection;
+import java.util.TreeSet;
+import java.text.Collator;
+
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.dom.DOMNode;
+import org.eclipse.lemminx.dom.LineIndentInfo;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.utils.XMLPositionUtility;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lemminx.extensions.contentmodel.model.CMElementDeclaration;
+import org.eclipse.lemminx.extensions.contentmodel.model.ContentModelManager;
+import org.eclipse.lemminx.extensions.contentmodel.model.CMDocument;
+import org.eclipse.lemminx.utils.StringUtils;
+import org.eclipse.lemminx.services.extensions.IPositionRequest;
+
+/**
+ * Code action to find the expected element name defined in the given xsd and
+ * replace it
+ * 
+ */
+public class cvc_elt_1_aCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+			SharedSettings sharedSettings, IComponentProvider componentProvider, CancelChecker cancelChecker) {
+
+		try {
+			DOMNode unexpectedElement = document.getDocumentElement();
+			if (unexpectedElement == null) {
+				return;
+			}
+			String unexpectedElementText = unexpectedElement.getNodeName();
+
+			// get range of text that needs to be replaced
+			int startOffset;
```

Comment:
Replace with:

```java
Range diagnosticRange = diagnostic.getRange();
int startOffset = document.offsetAt(diagnosticRange.getStart()) + 1;
DOMNode node = document.findNodeAt(startOffset);
if (node == null || !node.isElement()) {
	return;
}
Range replaceRange = XMLPositionUtility.selectStartTagName(node);
```

 - if I remember we need to use diagnostic range instead of code action range.
 - int startrOffset must be assigneddirectly.
 - replaceRange is computed only if needed.